processing/src/main/java/org/apache/druid/query/groupby/epinephelinae/GroupByQueryEngine.java

```diff
@@ -391,7 +391,7 @@ public boolean hasNext()
       if (delegate != null && delegate.hasNext()) {
         return true;
       } else {
-        if (!cursor.isDone() && granularizer.currentOffsetWithinBucket()) {
+        if (granularizer.currentOffsetWithinBucket()) {
```

Comment:
was the `cursor.isDone()` here removed simply because it's redundant?

Reply:
yea, i noticed granularizer.currentOffsetWithinBucket also checks isDone so it seemed nicer this way